### ISSUE TITLE
Select `relationship_table`.* is not added to select

### ISSUE THREAD

**Issue by Genaker** [kirschbaum-development/eloquent-power-joins#40]
I have a code: 

Models\CatalogProductEntity::leftJoinRelationship('catalog_product_entity_varchars.eav_attribute')->limit(1)->get();

and select is 
```
select `catalog_product_entity`.* from `catalog_product_entity` left join `catalog_product_entity_varchar` on `catalog_product_entity_varchar`.`row_id` = `catalog_product_entity`.`row_id` left join `eav_attribute` on `catalog_product_entity_varchar`.`attribute_id` = `eav_attribute`.`attribute_id` limit 1"
```
Select is : 

or even simple:
```
\Models\CatalogProductEntity::leftJoinRelationship('catalog_product_entity_varchars')->limit(1)->get();
```
Select Is:
```
"select `catalog_product_entity`.* from `catalog_product_entity` left join `catalog_product_entity_varchar` on `catalog_product_entity_varchar`.`row_id` = `catalog_product_entity`.`row_id` limit 1"
```
Should be 
```
"select `catalog_product_entity`.*, `catalog_product_entity_varchar`.*   from `catalog_product_entity` left join `catalog_product_entity_varchar` on `catalog_product_entity_varchar`.`row_id` = `catalog_product_entity`.`row_id` limit 1"
```
You have this described in the doc however you don't have doc how to add fields from the join. It doesn't have any sence join and not select fields:

https://github.com/kirschbaum-development/eloquent-power-joins#select--from-table

I have written some solution could you consider adding this to the core. and select with TableName.*
```
$query->leftJoinRelationship('eav_attribute')->addSelect(['eav_attribute.*']);
```

**Comment by luisdalmolin** [comment]
It is not the expected behavior to select the columns from the other table when joining. What the package does is basically check if there's a select statement set, and if not it sets `table.*` so you don't have conflicting fields which can cause some headaches. 

The solution is exactly as you added, or if you want to avoid power joins of adding the column automatically, you should call the add select first (but then you need to specify the two tables).

```php
$query->setSelect(['catalog_product_entity.*', 'eav_attribute.*'])->leftJoinRelationship('eav_attribute');
```

**Comment by stevethomas** [comment]
Sorry to revive this old issue, happy to open another issue or move to another medium if preferred; however this issue is pretty closed related. 

Some of my legacy code utilises joins on a query to the products table, and I am unable to produce the same performance using eager loading. 

I have also tried https://github.com/hammerstonedev/fast-paginate, and while this does speed up the main query once the offset gets above about 40k results, it is still time-consuming to eager load all the relationships, which mostly outweighs the benefits for the majority of the workload.

The way I have always approached joins is to alias the joined columns, like: 
```
select products.*, users.id AS users_id, users.username as users_username, files.id AS files_id files.path as files_path
from products
inner join users on products.user_id = users.id
inner join files on products.file_id = files.id
```

This might be a bit naive, but wouldn't there be a way to glue the related model into the product collection the same way eager/lazy loading does it? Obviously the related model will be missing a bunch of attributes (no worse than `->with('users.id,username')` though), and then you could easily read the relationship and access the model class. 

In other words; eager load via a join on one-to-one relationships and then hydrate the related model. I figure the author(s) of this package would be the people best placed to answer this 😉🤞🏻
